src/model.jl

```diff
@@ -332,18 +332,28 @@ function _add_constraints_demand_flexibility!(
     sets::Sets,
     demand_flexibility::DemandFlexibility,
     interval_length::Int,
+    init_shifted_demand::Array{Float64,1}=Float64[],
 )
     if demand_flexibility.rolling_balance
+        println("rolling load balance, first window: ", Dates.now())
+        JuMP.@constraint(
+            m,
+            rolling_load_balance_first[i in 1:sets.num_load_bus],
+            sum(
+                m[:load_shift_up][i, j] - m[:load_shift_dn][i, j] 
+                for j in 1:(demand_flexibility.duration - 1)
+            ) >= -1 * init_shifted_demand[i],
+        )
         println("rolling load balance: ", Dates.now())
         JuMP.@constraint(
             m,
             rolling_load_balance[
                 i in 1:sets.num_load_bus,
-                k in 1:(interval_length - demand_flexibility.duration)
+                k in 1:(interval_length - demand_flexibility.duration + 1)
```

Comment:
Nice catch.

Reply:
Should `k` start from 2 here?

Reply:
I think `k` should start at 1 here. I'm treating the net demand deviation as time `k = 0`, so the first rolling balance window (the separate constraint) accounts for that. Then the second window, which is included in this constraint, would start at `k = 1`

Reply:
Yeah, I see this is related our discussion above.